libkbfs/folder_branch_ops.go

```diff
@@ -1356,8 +1449,8 @@ func (fbo *folderBranchOps) SetInitialHeadFromServer(
 	// (e.g., calling `identifyOnce` and downloading the merged
 	// head) if head is already set.
 	lState := makeFBOLockState()
-	head := fbo.getHead(lState)
-	if head != (ImmutableRootMetadata{}) && head.mdID == md.mdID {
+	head, trusted := fbo.getHead(lState)
+	if trusted && head != (ImmutableRootMetadata{}) && head.mdID == md.mdID {
```

Comment:
compare explicitly to a `headTrustStatus` constant (will have to do once you change the underlying type to int)